From 10971fd2219843b2857c422075f58734beb01910 Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 8 Aug 2024 08:22:32 -0500
Subject: [PATCH] [TESTING] - EMMC04G-MK27 mmc quirk

20240717: Start: eMMC life_time: 0x01 0x01
20240722: 6.10.0-bone6: eMMC life_time: 0x01 0x01
20240808: Start: new board...

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 drivers/mmc/core/quirks.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/mmc/core/quirks.h b/drivers/mmc/core/quirks.h
index ddefedc9746e..cbdc1ec4793f 100644
--- a/drivers/mmc/core/quirks.h
+++ b/drivers/mmc/core/quirks.h
@@ -136,6 +136,16 @@ static const struct mmc_fixup __maybe_unused mmc_blk_fixups[] = {
 	MMC_FIXUP("M62704", CID_MANFID_KINGSTON, 0x0100, add_quirk_mmc,
 		  MMC_QUIRK_TRIM_BROKEN),
 
+	/*
+	 * Kingston EMMC04G-MK27 are showing fast wear rates
+	 * 2024???? : testing MMC_QUIRK_TRIM_BROKEN (eMMC died)
+	 * 20240708 : testing MMC_QUIRK_BROKEN_SD_DISCARD (eMMC failing)
+	 * 20240717 : testing MMC_QUIRK_TRIM_BROKEN | MMC_QUIRK_BROKEN_CACHE_FLUSH (eMMC died)
+	 * 20240808 : testing MMC_QUIRK_TRIM_BROKEN | MMC_QUIRK_BROKEN_SD_DISCARD | MMC_QUIRK_BROKEN_CACHE_FLUSH | MMC_QUIRK_ERASE_BROKEN (testing)
+	 */
+	MMC_FIXUP("MK2704", CID_MANFID_KINGSTON, 0x0100, add_quirk_mmc,
+		  MMC_QUIRK_TRIM_BROKEN | MMC_QUIRK_BROKEN_SD_DISCARD | MMC_QUIRK_BROKEN_CACHE_FLUSH | MMC_QUIRK_ERASE_BROKEN),
+
 	/*
 	 * Some SD cards reports discard support while they don't
 	 */
-- 
2.39.2

